engine/collection/compliance/engine_test.go

```diff
@@ -310,3 +311,23 @@ func (cs *ComplianceSuite) TestProcessUnsupportedMessageType() {
 	require.Error(cs.T(), err)
 	require.True(cs.T(), engine.IsIncompatibleInputTypeError(err))
 }
+
+// TestOnFinalizedBlock tests if finalized block gets processed when send through `Engine`.
+// Tests the whole processing pipeline.
+func (cs *ComplianceSuite) TestOnFinalizedBlock() {
+	finalizedBlock := unittest.ClusterBlockFixture()
+	cs.head = &finalizedBlock
+
+	done := atomic.NewBool(false)
+	cs.voteAggregator.On("PruneUpToView", finalizedBlock.Header.View).Run(func(mock.Arguments) {
+		done.Toggle()
+	}).Return(nil).Once()
+	cs.engine.OnFinalizedBlock(finalizedBlock.ID())
+
+	// matching engine has at least 100ms ticks for processing events
+	time.Sleep(1 * time.Second)
+
+	require.Eventually(cs.T(), done.Load, time.Second, time.Millisecond*20)
```

Comment:
Any reason we need both a sleep and an `Eventually` here? `Eventually` on its own will wait until the condition holds.
```suggestion

	require.Eventually(cs.T(), done.Load, time.Second, time.Millisecond*20)
```

Reply:
I forgot to remove sleep :(